server/service/live_queries.go

```diff
@@ -160,8 +223,16 @@ func (svc *Service) RunLiveQueryDeadline(
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			campaign, err := svc.NewDistributedQueryCampaign(ctx, "", &queryID, fleet.HostTargets{HostIDs: hostIDs})
+			queryIDPtr := &queryID
+			queryString := ""
+			// 0 is a special ID that indicates we should use raw SQL query instead
+			if queryID == 0 {
+				queryIDPtr = nil
+				queryString = query
+			}
+			campaign, err := svc.NewDistributedQueryCampaign(ctx, queryString, queryIDPtr, fleet.HostTargets{HostIDs: hostIDs})
 			if err != nil {
+				queryID = campaign.QueryID
```

Comment:
`campaign` or `campaign.QueryID` might not be set properly if `svc.NewDistributedQueryCampaign` failed.